sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -311,6 +312,97 @@ def predict(self, X):
         return X[:, 0]
 
 
+class OneClassErrorClassifier(BaseBadClassifier):
+    def fit(self, X, y):
+        # Convert data
+        X, y = check_X_y(
+            X, y, accept_sparse=("csr", "csc"), multi_output=True, y_numeric=True
+        )
+        cls, y = np.unique(y, return_inverse=True)
+        # find the number of class after trimming
+        if cls.shape[0] < 2:
+            raise ValueError("Nonsensical Error")
+
+    def predict(self, X):
+        X = check_array(X)
+        return np.ones(X.shape[0])
+
+
+class OneClassSampleErrorClassifier(BaseBadClassifier):
+    def fit(self, X, y, sample_weight=None):
+        # Convert data
+        X, y = check_X_y(
+            X, y, accept_sparse=("csr", "csc"), multi_output=True, y_numeric=True
+        )
+
+        cls, y = np.unique(y, return_inverse=True)
+        nb_cls = cls.shape[0]
+        if nb_cls < 2:
+            raise ValueError("normal class error")
+
+        # find the number of class after trimming
+        if sample_weight is not None:
+            nb_cls = cls.shape[0]
+            if len(sample_weight) > 0:
+                nb_cls = np.count_nonzero(np.bincount(y, sample_weight))
+            if nb_cls < 2:
+                raise ValueError("Nonsensical Error")
+
+        return self
+
+    def predict(self, X):
+        X = check_array(X)
+        return np.ones(X.shape[0])
+
+
+class OneClassSampleErrorClassifierPredict(BaseBadClassifier):
+    def __init__(self, flag=False):
+        self.flag = flag
+
+    def fit(self, X, y, sample_weight=None):
+        # Convert data
+        X, y = check_X_y(
+            X, y, accept_sparse=("csr", "csc"), multi_output=True, y_numeric=True
+        )
+
+        cls, y = np.unique(y, return_inverse=True)
+        nb_cls = cls.shape[0]
+        if nb_cls < 2:
+            raise ValueError("normal class error")
+
+        # find the number of class after trimming
+        if sample_weight is not None:
+            nb_cls = cls.shape[0]
+            if len(sample_weight) > 0:
+                nb_cls = np.count_nonzero(np.bincount(y, sample_weight))
+            if nb_cls < 2:
+                self.flag = True
+
+        return self
+
+    def predict(self, X):
+        X = check_array(X)
+        if self.flag:
+            return np.zeros(X.shape[0])
+        return np.ones(X.shape[0])
+
+
+def assert_raises_with_message_in_print(estimator, msg):
```

Comment:
When `OneClassSampleErrorClassifier` and `OneClassErrorClassifier` is tested we can use 

```python
from sklearn.utils._testing import raises
```

to check the message. Examples of how `raises` is are in `test_estimator_checks.py`.